packages/test/local-server-stress-tests/src/localServerStressHarness.ts

```diff
@@ -774,102 +787,49 @@ async function loadClient(
 }
 
 async function synchronizeClients(connectedClients: Client[]) {
-	const closeOrDispose = new Map<
-		Client,
-		{ error?: Error; rejects: ((reason?: any) => void)[] }
-	>();
-
-	const cleanUps: (() => void)[] = [];
-	for (const c of connectedClients) {
-		const rejector = (error) => {
-			const entry = closeOrDispose.get(c) ?? { rejects: [] };
-			entry.error ??= error;
-			closeOrDispose.set(c, entry);
-			entry.rejects.forEach((r) => r(entry.error));
-		};
-		c.container.once("closed", rejector);
-		c.container.once("disposed", rejector);
-		cleanUps.push(() => {
-			c.container.off("closed", rejector);
-			c.container.off("disposed", rejector);
-		});
-	}
-	try {
-		await Promise.all(
-			connectedClients.map(async (c) =>
-				timeoutPromise(
-					(resolve, reject) => {
-						const entry = closeOrDispose.get(c) ?? { rejects: [] };
-						closeOrDispose.set(c, entry);
-						if (entry.error) {
-							reject(entry.error);
-						} else if (c.container.connectionState !== ConnectionState.Connected) {
-							c.container.once("connected", () => resolve());
-							entry.rejects.push(reject);
-						} else {
-							resolve();
-						}
-					},
-					{
-						errorMsg: `Timed out waiting for client to connect ${c.tag}`,
-					},
-				),
-			),
-		);
-
-		await Promise.all(
-			connectedClients.map(async (c) =>
-				timeoutPromise(
-					(resolve, reject) => {
-						const entry = closeOrDispose.get(c) ?? { rejects: [] };
-						closeOrDispose.set(c, entry);
-						if (entry.error) {
-							reject(entry.error);
-						} else if (c.container.isDirty) {
-							c.container.once("saved", () => resolve());
-							entry.rejects.push(reject);
-						} else {
-							resolve();
-						}
-					},
-					{
-						errorMsg: `Timed out waiting for client to save ${c.tag}`,
-					},
-				),
-			),
-		);
-		const maxSeq = Math.max(
-			...connectedClients.map((c) => c.container.deltaManager.lastKnownSeqNumber),
-		);
-
-		const makeOpHandler = (c: Client, resolve: () => void, reject: (error) => void) => {
-			const entry = closeOrDispose.get(c) ?? { rejects: [] };
-			closeOrDispose.set(c, entry);
-			if (entry.error) {
-				reject(entry.error);
-			} else if (c.container.deltaManager.lastKnownSeqNumber < maxSeq) {
-				const handler = (msg) => {
-					if (msg.sequenceNumber >= maxSeq) {
-						c.container.off("op", handler);
-						resolve();
-					}
-				};
-				c.container.on("op", handler);
-				entry.rejects.push(reject);
-			} else {
+	return timeoutPromise((resolve, reject) => {
+		const handler = (error?) => {
```

Comment:
I like how cut and dry this strategy is - I wonder if `ensureSynchronized` in e2e tests could be simplified like this. Or if not... is this missing something?